sopel/privileges.py

```diff
@@ -1,146 +1,147 @@
+    .. important::
 
-    >>> priv >= OP
-    True
-    >>> bool(priv & HALFOP)
-    False
+        Do not refer directly to the integer value of a privilege level in your
+        code; the values may change. Use the appropriate member of this class as
+        a reference point instead.
 
-In that case, ``priv`` contains both VOICE and OP privileges, but not HALFOP.
-"""
-from __future__ import annotations
+    """
+    # values should behave as ints, but their string representations should
+    # still look like Enum
+    __str__ = enum.Enum.__str__
 
+    VOICE = enum.auto()
+    """Privilege level for the +v channel permission
 
-__all__ = [
-    'VOICE',
-    'HALFOP',
-    'OP',
-    'ADMIN',
-    'OWNER',
-    'OPER',
-]
+    .. versionadded:: 4.1
+    .. versionchanged:: 8.0
+
+        Constant moved from :mod:`sopel.plugin` to
+        :class:`sopel.privileges.AccessLevel`.
+
+    """
+
+    HALFOP = enum.auto()
+    """Privilege level for the +h channel permission
+
+    .. versionadded:: 4.1
+    .. versionchanged:: 8.0
+
+        Constant moved from :mod:`sopel.plugin` to
+        :class:`sopel.privileges.AccessLevel`.
 
+    .. important::
 
-VOICE = 1
-"""Privilege level for the +v channel permission
+        Not all IRC networks support this privilege mode. If you are writing a
+        plugin for public distribution, ensure your code behaves sensibly if
+        only ``+v`` (voice) and ``+o`` (op) modes exist.
 
-.. versionadded:: 4.1
-.. versionchanged:: 8.0
-   Moved into :mod:`sopel.privileges`.
-"""
+    """
 
-HALFOP = 2
-"""Privilege level for the +h channel permission
+    OP = enum.auto()
+    """Privilege level for the +o channel permission
 
-.. versionadded:: 4.1
-.. versionchanged:: 8.0
-   Moved into :mod:`sopel.privileges`.
+    .. versionadded:: 4.1
+    .. versionchanged:: 8.0
 
-.. important::
+        Constant moved from :mod:`sopel.plugin` to
+        :class:`sopel.privileges.AccessLevel`.
 
-    Not all IRC networks support this privilege mode. If you are writing a
-    plugin for public distribution, ensure your code behaves sensibly if only
-    ``+v`` (voice) and ``+o`` (op) modes exist.
+    """
 
-"""
+    ADMIN = enum.auto()
+    """Privilege level for the +a channel permission
 
-OP = 4
-"""Privilege level for the +o channel permission
+    .. versionadded:: 4.1
+    .. versionchanged:: 8.0
 
-.. versionadded:: 4.1
-.. versionchanged:: 8.0
-   Moved into :mod:`sopel.privileges`.
-"""
+        Constant moved from :mod:`sopel.plugin` to
+        :class:`sopel.privileges.AccessLevel`.
 
-ADMIN = 8
-"""Privilege level for the +a channel permission
+    .. important::
 
-.. versionadded:: 4.1
-.. versionchanged:: 8.0
-   Moved into :mod:`sopel.privileges`.
+        Not all IRC networks support this privilege mode. If you are writing a
+        plugin for public distribution, ensure your code behaves sensibly if
+        only ``+v`` (voice) and ``+o`` (op) modes exist.
 
-.. important::
+    """
```

Comment:
Nit: is it possible to express this as a cross-reference to the top-level description of why these flags may not be supported by all servers? Bit beyond my Sphinx-fu, but it seems like it might be slightly better to do `Not all IRC networks support this privilege mode, see <ref>`